project/src/main/java/com/google/autograder/servlets/helpers/API.java

```diff
@@ -26,26 +33,36 @@
 
 public final class API {
 
-    private static final String RESOURCES_PATH = "../../../project/src/main/resources"; // "../classes"
+    private static final String RESOURCES_PATH  = "../../../project/src/main/resources"; // "../classes"
+    private static final Logger LOGGER = Logger.getLogger(API.class.getName());
     private static final String API_KEY_PATH = "/auth/api_key.json";
     private static final Gson GSON = new Gson();
 
-    public static final String API_KEY = getAPIKey();
     public static final String UTF_8 = StandardCharsets.UTF_8.name();
+    public static final String API_KEY;
 
-    private static String getAPIKey() {
-        String api_key = null;
+    static {
+        API_KEY = retrieveAPIKey().orElse(null);
```

Comment:
If it's null, this should just throw an exception, otherwise you'll run into hard to debug exceptions later. It's typically better to have things fail faster (closer to the original source), as it'll be easier to debug

Reply:
+1